management-api-agent-5.0.x/src/main/java/com/datastax/mgmtapi/interceptors/CassandraDaemonInterceptor5x.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Please see the included license file for details.
+ */
+package com.datastax.mgmtapi.interceptors;
+
+import static com.datastax.mgmtapi.interceptors.CassandraDaemonInterceptor.socketFileStr;
+
+import com.datastax.mgmtapi.NodeOpsProvider;
+import com.datastax.mgmtapi.ShimLoader;
+import com.datastax.mgmtapi.ipc.IPCController;
+import com.google.common.collect.ImmutableMap;
+import io.k8ssandra.metrics.interceptors.MetricsInterceptor;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.epoll.Epoll;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicBoolean;
+import net.bytebuddy.agent.builder.AgentBuilder;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.implementation.MethodDelegation;
+import net.bytebuddy.implementation.SuperMethodCall;
+import net.bytebuddy.implementation.bind.annotation.SuperCall;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.ElementMatchers;
+import org.apache.cassandra.transport.CBUtil;
+import org.apache.cassandra.transport.Server;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CassandraDaemonInterceptor5x implements CassandraDaemonInterceptor {
+  private static final Logger logger = LoggerFactory.getLogger(CassandraDaemonInterceptor5x.class);
+  static final CassandraDaemonInterceptor instance = new CassandraDaemonInterceptor5x();
+  private static final AtomicBoolean isRunning = new AtomicBoolean(false);
+
+  private CassandraDaemonInterceptor5x() {}
+
+  public ElementMatcher<? super TypeDescription> type() {
+    return ElementMatchers.nameEndsWith(".CassandraDaemon");
+  }
+
+  public AgentBuilder.Transformer transformer() {
+    return (builder, typeDescription, classLoader, module, protectionDomain) ->
+        builder
+            .method(ElementMatchers.named("start"))
+            .intercept(
+                MethodDelegation.to(DefaultCassandraDaemonInterceptor.class)
+                    .andThen(MethodDelegation.to(MetricsInterceptor.class))
+                    .andThen(SuperMethodCall.INSTANCE));
+  }
+
+  public void intercept(@SuperCall Callable<Void> zuper) throws Exception {
+    try {
+      logger.info("Starting DataStax Management API Agent for Apache Cassandra v0.1");
+
+      NodeOpsProvider.instance.get().register();
+
+      if (!Epoll.isAvailable()) throw new RuntimeException("Event loop needed");
+
+      File unixSock = Paths.get(socketFileStr).toFile();
+
+      final EventLoopGroup group = new EpollEventLoopGroup(8);
+      final Server.ConnectionTracker connectionTracker = new _ConnectionTracker();
```

Comment:
This is the problematic bit. We can't call the new constructor that takes a BooleanSupplier function because it was marked private.